File: app/src/main/java/org/cuieney/videolife/presenter/contract/VideoHomeContract.java
package org.cuieney.videolife.presenter.contract;

import org.cuieney.videolife.common.base.BasePresenter;
import org.cuieney.videolife.common.base.BaseView;
import org.cuieney.videolife.entity.VideoListBean;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by cuieney on 17/2/24.
 */

public interface VideoHomeContract {

    interface View extends BaseView{
        void showContent(VideoListBean videoListBean);
        void error(Throwable throwable);
    }

    interface Presenter extends BasePresenter<View> {
        void getVideoData(String date);

    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/contract/VideoHomeContract.kt
package org.cuieney.videolife.kotlin.presenter.contract

import org.cuieney.videolife.entity.VeerListBean
import org.cuieney.videolife.entity.VideoListBean
import org.cuieney.videolife.kotlin.base.BasePresenter
import org.cuieney.videolife.kotlin.base.BaseView

/**
 * Created by cuieney on 2017/6/2.
 */
interface VideoHomeContract {
    interface View : BaseView {
        fun showContent(data: VideoListBean)
        fun error(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
        fun getVideoData(date: String, view: View)
    }
}